ethers/providers/jsonrpc/subscriptions.nim

```diff
@@ -77,14 +78,16 @@ method close*(subscriptions: JsonRpcSubscriptions) {.async, base.} =
     await subscriptions.unsubscribe(id)
 
 proc getCallback(subscriptions: JsonRpcSubscriptions,
-                 id: JsonNode): ?SubscriptionCallback =
+                 id: JsonNode): ?SubscriptionCallback  {. raises:[].} =
   try:
     if not id.isNil and id in subscriptions.callbacks:
-      subscriptions.callbacks[id].some
+        try:
+          return subscriptions.callbacks[id].some
+        except: discard
     else:
-      SubscriptionCallback.none
+      return SubscriptionCallback.none
   except KeyError:
-    SubscriptionCallback.none
+    return SubscriptionCallback.none
```

Comment:
spacing lines 84-86

What exception was leaking from `subscriptions.callbacks[id].some`? 

By default, SubscriptionCallback.none is returned, so we don't really need line 87-88.

The catching of `KeyError` is not needed with the try/except since it's caught on 86. If you're going to use try/except discard, then this whole thing can be a bit nicer:

```nim 
if not id.isNil and id in subscriptions.callbacks:
  try:
    return subscriptions.callbacks[id].some
  except: discard
```